app/src/androidTest/java/org/fossasia/pslab/MainActivityTest.java

```diff
@@ -0,0 +1,50 @@
+package org.fossasia.pslab;
+
+import android.support.test.espresso.action.ViewActions;
+import android.support.test.espresso.contrib.DrawerActions;
+import android.support.test.espresso.contrib.NavigationViewActions;
+import android.support.test.filters.LargeTest;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.fossasia.pslab.activity.MainActivity;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+
+/**
+ * Created by viveksb007 on 18/7/17.
+ */
+
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+public class MainActivityTest {
+
+    @Rule
+    public ActivityTestRule<MainActivity> mActivityRule = new ActivityTestRule<MainActivity>(MainActivity.class);
+
+    @Test
+    public void checkingAllFragments() throws InterruptedException {
+        Thread.sleep(2000);
+
+        onView(withId(R.id.drawer_layout)).perform(DrawerActions.open());
+        onView(withId(R.id.nav_view)).perform(NavigationViewActions.navigateTo(R.id.nav_applications));
+        Thread.sleep(1000);
```

Comment:
yes Instrumentation tests are connected test i.e they are performed on an emulator or an actual device. So if we won't provide the delay, there is a possibility that the intended view didn't came in foreground yet.